include/fastdds/dds/domain/DomainParticipant.hpp

```diff
@@ -547,6 +547,19 @@ class DomainParticipant : public Entity
             const std::string& profile_name,
             PublisherQos& qos) const;
 
+    /**
+     * Fills the PublisherQos with the first publisher profile found in the given XML (or the one specified).
+     *
+     * @param xml Raw XML string containing the profile to be used to fill the \c qos structure.
+     * @param qos PublisherQos object where the qos is returned.
+     * @param profile_name Publisher profile name. Empty by default (first one found).
+     * @return RETCODE_OK on success. RETCODE_BAD_PARAMETER otherwise.
+     */
+    FASTDDS_EXPORTED_API ReturnCode_t get_publisher_qos_from_xml(
```

Comment:
We decided to have at least two separate functions.

An overload with `profile_name` argument. This function will expects a complete Fast DDS XML and will retrieve the profile specified. In case `profile_name` is empty, might return the default one (this can be done in another `get_default_...` function. If the profile is not found, return error.

An overload function without `profile_name` argument. This XML might not follow the Fast DDS XML structure and will retrieve the first profile found.